src/util/bip32.rs

```diff
@@ -210,12 +210,31 @@ impl serde::Serialize for ChildNumber {
     }
 }
 
+/// Trait that allows possibly failable conversion from a type into a
+/// derivation path
+pub trait IntoDerivationPath {
```

Comment:
So, this is basically just `TryInto<DerivationPath, Error=Error>`? Wish we could just use it directly (1.34.0), but if we can't maybe it would be better to just name it `TryIntoDerivationPath` to make the future transition easier? There might be some little kinks wrt such a transition, but since `rust-bitcoin` can still do breaking changes on ongoing basis, there's no point in worrying too much about it.

Reply:
Yes, if we were >=1.34, it would be `TryInto`. Regarding current naming, many Rust `Into/FromSomething` have the name without `Try` even if they fail (`FromStr` for instance) and I do not know any counter-examples of traits named with `Try...` other than `TryInto/From` themselves. Also this will change method naming, which will hardly derive from the naming used across this crate (where `from_...` usually failiable)